plugin/pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -506,6 +506,269 @@ func TestForgetPod(t *testing.T) {
 	}
 }
 
+func addResource(r *Resource, rl v1.ResourceList) {
+	if r == nil {
+		return
+	}
+
+	for rName, rQuant := range rl {
+		switch rName {
+		case v1.ResourceCPU:
+			r.MilliCPU += rQuant.MilliValue()
+		case v1.ResourceMemory:
+			r.Memory += rQuant.Value()
+		case v1.ResourceNvidiaGPU:
+			r.NvidiaGPU += rQuant.Value()
+		default:
+			if v1helper.IsOpaqueIntResourceName(rName) {
+				r.AddOpaque(rName, rQuant.Value())
+			}
+		}
+	}
+}
+
+func getResourceRequest(pod *v1.Pod) v1.ResourceList {
+	result := &Resource{}
+	for _, container := range pod.Spec.Containers {
+		addResource(result, container.Resources.Requests)
+	}
+
+	return result.ResourceList()
+}
+
+func getPort(pod *v1.Pod) map[int]bool {
```

Comment:
Is this getUsedPorts? I may be wrong - I guess we already have a similar fn in metadata.go in predicates? Can we re-use it here or move it to some other package so that we can import here? WDYT?